delft/sequenceLabelling/wrapper.py

```diff
@@ -157,28 +152,30 @@ def train(self, x_train, y_train, f_train=None, x_valid=None, y_valid=None, f_va
 
         if not y_valid is None:
             y_all = np.concatenate((y_train, y_valid), axis=0)
-        else: 
+        else:
             y_all = y_train
 
         features_all = concatenate_or_none((f_train, f_valid), axis=0)
 
         self.p = prepare_preprocessor(x_all, y_all, features=features_all, model_config=self.model_config)
-        if self.bert_preprocessor != None:
-            self.bert_preprocessor.set_empty_features_vector(self.p.empty_features_vector())
-            self.bert_preprocessor.set_empty_char_vector(self.p.empty_char_vector())
+        if self.transformer is not None:
+            self.transformer.init_preprocessor(self.model_config.max_sequence_length,
+                                               empty_features_vector=self.p.empty_features_vector(),
+                                               empty_char_vector=self.p.empty_char_vector())
 
         self.model_config.char_vocab_size = len(self.p.vocab_char)
         self.model_config.case_vocab_size = len(self.p.vocab_case)
 
-        self.model = get_model(self.model_config, self.p, len(self.p.vocab_tag), load_pretrained_weights=True)
+        self.model = get_model(self.model_config, self.p, len(self.p.vocab_tag), load_pretrained_weights=True,
+                               transformer=self.transformer)
         trainer = Trainer(self.model,
                           self.models,
                           self.embeddings,
                           self.model_config,
                           self.training_config,
                           checkpoint_path=self.log_dir,
-                          preprocessor=self.p, 
-                          bert_preprocessor=self.bert_preprocessor
+                          preprocessor=self.p,
+                          transformer=self.transformer
```

Comment:
The advantage of passing just the transformer preprocessor is that we can get rid of the transformer object.